### PR TITLE
Precise search for Similar Tracks

### PR DESCRIPTION
**Pull request by Eiron**
Discovered issues with some of the selections of Similar Tracks in appropriate playlists (e.g. *Nirvana - Lithium (Live)* when *Nirvana - Lithium* is available). Copying - hopefully - for similar functionality as other library searches.

### CODE REVIEW

**Comment by URY-GitHub** [comment]
Can one of the admins verify this patch?

**Comment by LloydW93** [comment]
Should be good once you resolve merge conflicts.

File: src/Classes/ServiceAPI/MyRadio_Track.php [hunk 1]
```diff
@@ -770,10 +770,21 @@ public function getSimilar() {
             }
             foreach ($data['similartracks']['track'] as $r) {
                 if ($r['match'] >= 0.25) {
+                    //Try to find an exact match
                     $c = self::findByOptions(['title' => $r['name'],
                                 'artist' => $r['artist']['name'],
                                 'limit' => 1,
-                                'digitised' => true]);
+                                'digitised' => true
```
Comment (AJDurant) [inline src/Classes/ServiceAPI/MyRadio_Track.php]: needs a comma to separate elements in array

File: src/Classes/ServiceAPI/MyRadio_Track.php [hunk 2]
```diff
@@ -770,10 +770,21 @@ public function getSimilar() {
...
+                                'precise' => true]);
+                    //Try to find a not-so-exact match
+                    if (!empty($c)) {
```
Comment (AJDurant) [inline src/Classes/ServiceAPI/MyRadio_Track.php]: if not empty is not the right condition to match. It probably will be empty if there are no matches, though that should be checked what findByOptions returns

File: src/Classes/ServiceAPI/MyRadio_Track.php [hunk 3]
```diff
@@ -770,10 +770,21 @@ public function getSimilar() {
...
+                        $c = self::findByOptions(['title' => $r['name'],
+                                'artist' => $r['artist']['name'],
+                                'limit' => 1,
+                                'digitised' => true
+                                'precise' => false]);
+                    }
+                    //Still nothing
```
Comment (AJDurant) [inline src/Classes/ServiceAPI/MyRadio_Track.php]: This comment is not true, the next condition only runs if there is something (it adds the found track to the array lastfm_similar)
Better comment would be something like: Add the track to the similar list if we have it

**Comment by Eiron** [comment]
Yep. I see those errors. Not sure how they fill in. I copy-pasted the code, but must have made some weird errors somewhere along the line somehow. This version should fix those errors? Let me know if it needs any more repairs before it can be put live.

**Comment by AJDurant** [comment]
you need to fix the merge conflicts, run 'git merge master' on your branch ad then fix anything that cannot be automatically resolved

**Comment by AJDurant** [comment]
replaced by #452

**Pull request closed without merging**